vhdbuilder/prefetch/internal/components/types.go

```diff
@@ -0,0 +1,25 @@
+package components
+
+// List represents a container image component list (e.g. parts/linux/cloud-init/artifacts/components.json).
+type List struct {
+	Images []Image `json:"ContainerImages,omitempty"`
+}
+
+// Image represents a container image component. Note that these are the only fields
+// we need from container image component definitions to generate prefetch scripts.
+type Image struct {
+	DownloadURL       string `json:"downloadURL"`
+	MultiArchVersions []struct {
+		LatestVersion         string `json:"latestVersion"`
+		PreviousLatestVersion string `json:"previousLatestVersion"`
```

Comment:
Do we not need to add `omitempty` for an optional field? Maybe Go has handled the optional case by default. If you tested it, it's fine.
1 example I am referring to. https://stackoverflow.com/questions/34053815/handle-optional-json-field-in-http-request-body

Reply:
`omitempty` is only really useful for ptr types in this case `ContainerImagePrefetchOptimization` is a value type, where the Binaries property (which is a reference type) already has an `omitempty`

we can add it to be safe, though I don't _think_ it will make a difference

Reply:
oh I see, you meant to align with what you have in your new schema - I think it's the same case though (`string` is a value type rather than a ptr type)

Reply:
updated